src/lib/sql/sql_translator.cpp

```diff
@@ -757,14 +757,26 @@ void SQLTranslator::_translate_select_list_groupby_having(const hsql::SelectStat
     visit_expression(having_expression, find_aggregates_and_arguments);
   }
 
-  // Build pre_aggregate_projection, i.e. evaluate all Expression required for GROUP BY/Aggregates
-  if (pre_aggregate_expressions.size() != _current_lqp->column_expressions().size()) {
-    _current_lqp = ProjectionNode::make(pre_aggregate_expressions, _current_lqp);
-  }
+  const auto is_aggregate = !aggregate_expressions.empty() || !group_by_expressions.empty();
+
+  const auto pre_aggregate_lqp = _current_lqp;
 
   // Build Aggregate
-  const auto is_aggregate = !aggregate_expressions.empty() || !group_by_expressions.empty();
   if (is_aggregate) {
+    // Build pre_aggregate_projection, i.e. evaluate all Expression required for GROUP BY/Aggregates
```

Comment:
Few points for this block: I know it's c/p, but there is no variable "pre_aggregate_projection", but the comment suggests so.

```c++
const auto aggregate_argument_required_computation = std::any_of(pre_aggregate_expressions.begin(), pre_aggregate_expressions.end(), [&](const auto& aggregate_argument) {
  return !_current_lqp->find_column_id(*expression);
});
```

would be shorter and clearer to me.

Reply:
Fixed. Continued to use `any_expression_not_yet_available` instead of `aggregate_argument_required_computation` because we are not only talking about aggregate arguments, but also group by expressions